src/lib/CMakeLists.txt

```diff
@@ -5,6 +5,7 @@ set(
     abstract_expression.hpp
```

Comment:
Continuing the discussion from https://github.com/hyrise/hyrise/pull/673#discussion_r167011790

I went for @mrks template magic and adjusted it just slightly. On second reading it didn't appear that obscure at all and it works like a charm for everything except CreateViewNode (which takes an LQP as its last constructor param, which make can't deal with)

Reply:
We could probably use `std::is_constructible` to check if the LQP argument at the end belongs to the constructor or if it is a child.

Reply:
template metaprogramming is just amazing....

Reply:
I don't wanna do this. This would require doing is_constructible for all but the last, and all but the two last params and that just gets too much for now, imo.

Reply:
What is the current solution for CreateViewNode? Not use make?

Reply:
jenau

Reply:
ok, that works for now. I can't think of any other case where we need an LQP as a constructor parameter for another node.